Index_finger_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_0_0
    - joint_1_0
    - joint_2_0
    - joint_3_0
  gains:
    joint_0_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_1_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_2_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_3_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
Middle_finger_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_4_0
    - joint_5_0
    - joint_6_0
    - joint_7_0
  gains:
    joint_4_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_5_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_6_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_7_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
Ring_finger_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_8_0
    - joint_9_0
    - joint_10_0
    - joint_11_0
  gains:
    joint_8_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_9_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_10_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_11_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
Thumb_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_12_0
    - joint_13_0
    - joint_14_0
    - joint_15_0
  gains:
    joint_12_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_13_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_14_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_15_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
HAND_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_0_0
    - joint_1_0
    - joint_2_0
    - joint_3_0
    - joint_12_0
    - joint_13_0
    - joint_14_0
    - joint_15_0
    - joint_4_0
    - joint_5_0
    - joint_6_0
    - joint_7_0
    - joint_8_0
    - joint_9_0
    - joint_10_0
    - joint_11_0
  gains:
    joint_0_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_1_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_2_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_3_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_12_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_13_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_14_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_15_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_4_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_5_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_6_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_7_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_8_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_9_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_10_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    joint_11_0:
      p: 100
      d: 1
      i: 1
      i_clamp: 1